### PR TITLE
feat: create friend links between users 

### PR DESCRIPTION
**Pull request by Fyuryy**
# What I did
Created functions to add and remove friends 

# How I did it
Modified the user file to create the new two functions. Linked the functions to the place where they have to be called in ExternalProfileScreen and tested the globality of the code. 

# How to verify it
Try to add and/or remove a friend in your profile 

# Demo video
https://github.com/uniconnect-epfl/uniconnect/assets/91114548/a77bf4e0-1b08-4e41-bf04-4950ed57599b

# Pre-merge checklist
The changes I have introduced:
- [ ] work correctly
- [ ] do not break other functionalities
- [ ] work correctly on Android
- [ ] are fully tested

### CODE REVIEW

**Review by GustaveCharles — CHANGES_REQUESTED** [review]
LGTM.However, I have not been able to test the functionalities of the PR so I cannot approve it yet.

File: __test__/screens/QrScan/QrScanScreen.test.tsx
[the remark this answers is not in the document]
Comment (Fyuryy) [reply __test__/screens/QrScan/QrScanScreen.test.tsx:1]: Yes indeed. There was a wild "console.log(...)" that made all tests fail. I had to capture it!

File: __test__/firebase/User.test.ts
[the remark this answers is not in the document]
Comment (Fyuryy) [reply __test__/firebase/User.test.ts:1]: Thanks!

File: screens/Profile/ExternalProfileScreen/ExternalProfileScreen.tsx
[the remark this answers is not in the document]
Comment (Fyuryy) [reply screens/Profile/ExternalProfileScreen/ExternalProfileScreen.tsx:1]: I don't think I have understood your comment... what exactly do you mean?

**Review by GustaveCharles — APPROVED** [review]

**Pull request merged**